File: SuperiorPackGroup/BLL/ProjectsBLL.vb
Imports DevExpress.Xpo
Imports DXDAL.SPGData
Imports System.ComponentModel
Imports System.Linq

Public Class ProjectsBLL

    Public Shared Function GetProjectDetailChanges(ByVal session As Session, ByVal recordID As Integer) As BindingList(Of ChangeObject)

        Dim result As New List(Of ChangeObject)

        For Each detail As ProjectDetails In session.GetObjectByKey(Of Project)(recordID).ProjectDetails
            result.AddRange(AuditTrailBLL.GetChanges(session, "ProjectDetails", detail.Oid, True).ToList())
        Next

        Return New BindingList(Of ChangeObject)(result)

    End Function

    Public Shared Function ValidateRecord(ByVal ProjectName As String, ByVal CustomerID As Integer?, ByVal ReqStartDate As Date?, ByVal ReqDeliveryDate As Date?, ByVal ReqShipDate As Date?) As Boolean

        If ReqDeliveryDate.HasValue AndAlso ReqDeliveryDate.Value = #12:00:00 AM# Then
            ReqDeliveryDate = Nothing
        End If

        If ReqShipDate.HasValue AndAlso ReqShipDate.Value = #12:00:00 AM# Then
            ReqShipDate = Nothing
        End If

        If String.IsNullOrEmpty(ProjectName) OrElse CustomerID.HasValue = False OrElse (ReqDeliveryDate.HasValue = False AndAlso ReqShipDate.HasValue = False) Then
            Return False
        End If

        If ReqStartDate.HasValue AndAlso ((ReqShipDate.HasValue AndAlso ReqStartDate.Value > ReqShipDate.Value) OrElse (ReqDeliveryDate.HasValue AndAlso ReqStartDate.Value.Date > ReqDeliveryDate.Value.Date)) Then
            Return False
        End If

        Return True

    End Function



End Class

File: SuperiorPackGroup/BLL/ProjectsBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using DevExpress.Xpo;
using DXDAL.SPGData;
using System.ComponentModel;
using System.Linq;

namespace SuperiorPackGroup
{
	public class ProjectsBLL
	{

		public static BindingList<ChangeObject> GetProjectDetailChanges(Session session, int recordID)
		{

			List<ChangeObject> result = new List<ChangeObject>();

			foreach (ProjectDetails detail in session.GetObjectByKey<Project>(recordID).ProjectDetails)
			{
				result.AddRange(AuditTrailBLL.GetChanges(session, "ProjectDetails", detail.Oid, true).ToList());
			}

			return new BindingList<ChangeObject>(result);

		}

		public static bool ValidateRecord(string ProjectName, int? CustomerID, DateTime? ReqStartDate, DateTime? ReqDeliveryDate, DateTime? ReqShipDate)
		{

			if (ReqDeliveryDate.HasValue && ReqDeliveryDate.Value == Convert.ToDateTime("12:00:00 AM"))
			{
				ReqDeliveryDate = null;
			}

			if (ReqShipDate.HasValue && ReqShipDate.Value == Convert.ToDateTime("12:00:00 AM"))
			{
				ReqShipDate = null;
			}

			if (string.IsNullOrEmpty(ProjectName) || CustomerID.HasValue == false || (ReqDeliveryDate.HasValue == false && ReqShipDate.HasValue == false))
			{
				return false;
			}

			if (ReqStartDate.HasValue && ((ReqShipDate.HasValue && ReqStartDate.Value > ReqShipDate.Value) || (ReqDeliveryDate.HasValue && ReqStartDate.Value.Date > ReqDeliveryDate.Value.Date)))
			{
				return false;
			}

			return true;

		}



	}

}